core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -1607,6 +1607,10 @@ private RecordCursorFactory generateTableQuery(
                 throw e;
             }
 
+            if (model.getJoinType() == QueryModel.JOIN_ASOF || model.getJoinType() == QueryModel.JOIN_CROSS) {
```

Comment:
could we please cache branch outcome in boolean flag and then "pop" in "finally" block.